packages/core/http/core-http-server/src/router/scratch_pad.ts

```diff
@@ -0,0 +1,190 @@
+  handler: RequestHandler<P, Q, B, Context, Method>
+) => void;
+
+const examplePostRegistrar1: NewRouteRegistrar1<unknown, unknown, unknown> = {} as any;
+
+/**
+ * EXAMPLE USAGE:
+ */
+
+const common = {
+  path: '/api/my-plugin/my-route',
+  options: {},
+};
+
+examplePostRegistrar1(
+  {
+    ...common,
+    version: '1',
+    validate: { body: schema.object({ name: schema.string() }) },
+  },
+  async (ctx, req, res) => res.ok()
+);
+
+examplePostRegistrar1(
+  {
+    ...common,
+    version: '2',
+    validate: { body: schema.object({ name: schema.string(), lastName: schema.string() }) },
+  },
+  async (ctx, req, res) => res.ok()
+);
+
+examplePostRegistrar1(
+  {
+    ...common,
+    validate: {
+      body: schema.object({
+        name: schema.string(),
+        lastName: schema.string(),
+        age: schema.number(),
+      }),
+    },
+    version: '3',
+  },
+  async (ctx, req, res) => res.ok()
+);
+
+/**
+ * ===================== End first proposal =====================
+ */
+
+/**
+ * ===================== Second proposal =====================
+ *
+ * See "Current API". We could also:
+ *
+ * Change: 3, 5
+ * Keep constant: 1, 2, 4
+ *
+ * If 3 & 5 change across versions it implies a coupling between these two. Specifically,
+ * whenever a new version is added we know we need new validation (for the new API) and
+ * we need to tite this to a handler.
+ *
+ * One approach could be to tightly pair the validation and handler in the registrar's
+ * API.
+ */
+
+interface Handler<
+  P,
+  Q,
+  B,
+  Context extends RequestHandlerContextBase = RequestHandlerContextBase,
+  Method extends RouteMethod = 'get'
+> {
+  validate: RouteValidatorFullConfig<P, Q, B> | false;
+  handler: RequestHandler<P, Q, B, Context, Method>;
+}
+
+type NewRouteRegistrar2<
+  Method extends RouteMethod = 'get',
+  Context extends RequestHandlerContextBase = RequestHandlerContextBase
+> = (
+  route: Omit<RouteConfig<unknown, unknown, unknown, Method>, 'validate'>,
+  handlers: {
+    [version in Version]?: Handler<unknown, unknown, unknown, Context, Method>;
+  }
+) => void;
+
+const examplePostRegistrar2: NewRouteRegistrar2 = {} as any;
+
+/**
+ * EXAMPLE USAGE:
+ */
+
+const v1Route: Handler<unknown, unknown, unknown> = {
+  validate: {
+    body: schema.object({
+      name: schema.string(),
+    }),
+  },
+  handler: async (ctx, req, res) => res.ok(),
+};
+
+const v2Route: Handler<unknown, unknown, unknown> = {
+  validate: {
+    body: schema.object({
+      name: schema.string(),
+      lastName: schema.string(),
+    }),
+  },
+  handler: async (ctx, req, res) => res.ok(),
+};
+
+examplePostRegistrar2(
+  {
+    path: '/api/my-plugin/my-route',
+    options: {},
+  },
+  {
+    '1': v1Route,
+    '2': v2Route,
+  }
+);
+
+/**
+ * ===================== End second proposal =====================
+ */
+
+/**
+ * Open questions:
+ *
+ * 1. What is the default behaviour when we don't have a version specified? Latest stable?
```

Comment:
Defaulting to the latest stable version makes sense to me.

Reply:
That makes sense to me too. I added this question because I believe we will need to follow what the other clients/API behaviors will be doing. I'll move this to another section!